bot/exts/utilities/challenges.py

```diff
@@ -0,0 +1,309 @@
+            name="`Total Attempts`",
+            value=f"**{kata_information['totalAttempts']}**",
+            inline=False,
+        )
+        return miscellaneous_embed
+
+    @staticmethod
+    def create_view(dropdown: InformationDropdown, link: str) -> ui.View:
+        """
+        Creates the 'view', which is essentially a name for the discord UI.
+
+        The discord UI is implemented onto the embed, where the user can choose what information about the kata they
+        want, along with a link button to the kata itself.
+        """
+        view = ui.View()
+        # adding the dropdown
+        view.add_item(dropdown)
+        # manually creating the button as it is a link button and has no decorator
+        view.add_item(ui.Button(label="View the Kata", url=link))
+        return view
+
+    @commands.command(aliases=["kata"])
+    async def challenge(self, ctx: commands.Context, language: str = "python", *, query: str = None) -> None:
+        """
+        The challenge command pulls a random kata (challenge) from codewars.com.
+
+        The different ways to use this command are:
+        `.challenge <language>` - Pulls a random challenge within that language's scope.
+        `.challenge <language> <difficulty>` - The difficulty can be from 1-8,
+        1 being the hardest, 8 being the easiest. This pulls a random challenge within that difficulty & language.
+        `.challenge <language> <query>` - Pulls a random challenge with the query provided under the language
+        `.challenge <language> <query>, <difficulty>` - Pulls a random challenge with the query provided,
+        under that difficulty within the language's scope.
+        """
+        get_kata_link = f"https://codewars.com/kata/search/{language}"
+
+        if language and not query:
+            level = f"-{choice(['1', '2', '3', '4', '5', '6', '7', '8'])}"
+        elif "," in query or ", " in query:
+            query_splitted = query.split("," if ", " not in query else ", ")
+
+            if len(query_splitted) > 2:
+                raise commands.BadArgument(
+                    "There can only be one comma within the query, separating the difficulty and the query itself."
+                )
+
+            query, level = query_splitted
+            level = f"-{level}"
+        elif query.isnumeric():
+            level, query = f"-{query}", None
+        else:
+            level = None
+
+        params = {name: value for name, value in zip(["q", "r[]"], [query, level]) if value}
+        params = {**params, "beta": "false"}
+
+        first_kata_id = await self.kata_id(get_kata_link, params)
+
+        kata_information = await self.kata_information(first_kata_id)
+
+        kata_embed = self.main_embed(kata_information)
+
+        language_embed = self.language_embed(kata_information)
+
+        tags_embed = self.tags_embed(kata_information)
+
+        miscellaneous_embed = self.miscellaneous_embed(kata_information)
```

Comment:
I appreciate the social distancing on this one, but we still need to build up a herd-immunity.
```suggestion
        first_kata_id = await self.kata_id(get_kata_link, params)
        kata_information = await self.kata_information(first_kata_id)

        kata_embed = self.main_embed(kata_information)
        language_embed = self.language_embed(kata_information)
        tags_embed = self.tags_embed(kata_information)
        miscellaneous_embed = self.miscellaneous_embed(kata_information)
```

Reply:
Haha, yeah I'll be fixing all of these when I get back to my device